sycl/test/host-interop-task/host-task.cpp

```diff
@@ -55,10 +80,159 @@ void test2() {
       assert(Acc[Idx] == 123);
     }
   }
+
+  Q.wait_and_throw();
+}
+
+// Host-task depending on another host-task via both buffers and
+// handler::depends_on() should not hang
+void test3() {
+  queue Q(EH);
+
+  static constexpr size_t BufferSize = 10 * 1024;
+
+  buffer<int, 1> B0{range<1>{BufferSize}};
+  buffer<int, 1> B1{range<1>{BufferSize}};
+  buffer<int, 1> B2{range<1>{BufferSize}};
+  buffer<int, 1> B3{range<1>{BufferSize}};
+  buffer<int, 1> B4{range<1>{BufferSize}};
+  buffer<int, 1> B5{range<1>{BufferSize}};
+  buffer<int, 1> B6{range<1>{BufferSize}};
+  buffer<int, 1> B7{range<1>{BufferSize}};
+  buffer<int, 1> B8{range<1>{BufferSize}};
+  buffer<int, 1> B9{range<1>{BufferSize}};
+
+  std::vector<event> Deps;
+
+  for (size_t Idx = 0; Idx < 10; ++Idx) {
+    event E = Q.submit([&](handler &CGH) {
+      CGH.depends_on(Deps);
+
+      std::cout << "Submit: " << Idx << std::endl;
+
+      auto Acc0 = B0.get_access<mode::read_write, target::host_buffer>(CGH);
+      auto Acc1 = B1.get_access<mode::read_write, target::host_buffer>(CGH);
+      auto Acc2 = B2.get_access<mode::read_write, target::host_buffer>(CGH);
+      auto Acc3 = B3.get_access<mode::read_write, target::host_buffer>(CGH);
+      auto Acc4 = B4.get_access<mode::read_write, target::host_buffer>(CGH);
+      auto Acc5 = B5.get_access<mode::read_write, target::host_buffer>(CGH);
+      auto Acc6 = B6.get_access<mode::read_write, target::host_buffer>(CGH);
+      auto Acc7 = B7.get_access<mode::read_write, target::host_buffer>(CGH);
+      auto Acc8 = B8.get_access<mode::read_write, target::host_buffer>(CGH);
+      auto Acc9 = B9.get_access<mode::read_write, target::host_buffer>(CGH);
+
+      CGH.codeplay_host_task([=] {
+        uint64_t X = 0;
+
+        X ^= reinterpret_cast<uint64_t>(&Acc0[Idx + 0]);
+        X ^= reinterpret_cast<uint64_t>(&Acc1[Idx + 1]);
+        X ^= reinterpret_cast<uint64_t>(&Acc2[Idx + 2]);
+        X ^= reinterpret_cast<uint64_t>(&Acc3[Idx + 3]);
+        X ^= reinterpret_cast<uint64_t>(&Acc4[Idx + 4]);
+        X ^= reinterpret_cast<uint64_t>(&Acc5[Idx + 5]);
+        X ^= reinterpret_cast<uint64_t>(&Acc6[Idx + 6]);
+        X ^= reinterpret_cast<uint64_t>(&Acc7[Idx + 7]);
+        X ^= reinterpret_cast<uint64_t>(&Acc8[Idx + 8]);
+        X ^= reinterpret_cast<uint64_t>(&Acc9[Idx + 9]);
+
+        std::cout << "  Start " << Idx << " (" << X << ")" << std::endl;
+        using namespace std::chrono_literals;
+        std::this_thread::sleep_for(100ms);
```

Comment:
Why sleep is needed here?

Reply:
It was originally in customer's code. Though, I don't really know if it's needed here.

Reply:
Removed